### ISSUE TITLE
Support wildcards in template policy resources

### ISSUE THREAD

**Issue by WanderingStar** [cedar-policy/cedar#95]
### Category

Cedar language features

### Describe the feature you'd like to request

We would like to be able to use template linking to grant permissions on patterns of resources, for example "Shohreh has the Editor policy on all Documents matching *.pdf".

This depends on #81 

### Describe the solution you'd like

Allow the `like` operator to accept expressions, not just literal strings on the RHS. Support like on entities, with the logic that the LHS entity must be the same EntityType as the RHS entity and the LHS entity's name must be `like` the RHS entity.

```
@id("Editor")
permit(
    principal == ?principal,
    action in [Action::"Read", Action::"Write", Action::"Comment"],
    resource like ?resource
);
```

### Describe alternatives you've considered

To achieve this goal today, we can create more policies with the wildcards in literal strings, but we would prefer to use policy templates so that the templates can evolve.

If #94 is implemented, one option would be to separate the type check from the name check: 
```
resource.name like ?resource.name
&& resource is Document
```

If there were more placeholders than just `?principal` and `?resource`, maybe a string placeholder would be better:
```
resource.name like ?pattern
&& resource is Document
```

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by khieta** [comment]
> A "Role" becomes a policy template, and granting the role is creating a template link.

Yes, I think this is a natural way to think about things. Another way to do this is to use the entity hierarchy to help express roles. For example, you could say that every user in `UserGroup::"roleA"` is allowed to read/write all files with a name like `*.doc` using the following policy:
```
permit(
    principal in UserGroup::"roleA",
    action in [Action::"Read", Action::"Write"],
    resource
) when {
    resource.filename like "*.doc"
};
```
Then when you want to give a user `User::"alice"` this "role", you can add `UserGroup::"alice"` to be a member of `UserGroup::"roleA"` in your entity hierarchy.

You could also do something similar to group files together. For example, you could create an entity `FileGroup::"doc"` that will be the parent of all files with an entity ids like `*.doc`, and then use a template like 
```
permit(
    principal in ?principal,
    action in [Action::"Read", Action::"Write"],
    resource in ?resource
);
```
where you instantiate `?principal` to be `UserGroup::"foo"` (for example) and `?resource` to be `FileGroup::"doc"`.

But getting back to your original question: If you're ok with the "filename" solution, then I think you'll want to file a new issue that requests allowing templates to be instantiated with non-entity type values (`String`s in this case). If the "filename" solution won't work for you, then you could consider the group-based approach above, or narrow the scope of the current issue to ask for allowing `like` operations on entity ids.

**Comment by WanderingStar** [comment]
Seems like the RFC process is a better way to deal with this: https://github.com/cedar-policy/rfcs

**Comment by khieta** [comment]
Closing this issue in favor of [rfc#3](https://github.com/cedar-policy/rfcs/pull/3) (formerly issue #81) and #119.
